### PR TITLE
sile: use `lpeg` formula

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Follow-up to #130960.

### CODE REVIEW

**Review by SMillerDev — APPROVED** [review]

**Comment by cho-m** [comment]
The metadata is installed by `luarocks` so you would need to change `lpeg` formula to install via `luarocks`. Though not sure on the multiple Rocks trees.

Alternatively, you can try `--deps-mode=none` to ignore missing `lpeg`. May need to also add `#{Formula["lpeg"].opt_share}/lua/#{luaversion}/?.lua` to `LUA_PATH`.

**Comment by chenrui333** [comment]
looks like the build passed, but just never got reviewed and merged?

File: Formula/sile.rb [hunk 1]
```diff
@@ -142,13 +138,18 @@ def install
     luapath = libexec/"vendor"
 
     paths = %W[
+      #{Formula["lpeg"].opt_share}/lua/#{luaversion}/?.lua
```
Comment (MikeMcQuaid) [inline Formula/sile.rb:141]: Should this have a `?`?

**Review by MikeMcQuaid — APPROVED** [review]

File: Formula/sile.rb (hunk 1 above)
Comment (carlocab) [reply Formula/sile.rb:141]: Yes, the `?` is a placeholder for arbitrary file basenames.

**Pull request closed without merging**